libs/core/tests/unit_tests/utils/test_function_calling.py

```diff
@@ -69,6 +69,22 @@ def test_convert_to_openai_function(
         },
     }
 
-    for fn in (pydantic, function, tool, expected):
+    for fn in (pydantic, function, dummy_tool, expected):
         actual = convert_to_openai_function(fn)  # type: ignore
         assert actual == expected
+
+
+@pytest.mark.xfail(reason="Pydantic converts Optional[str] to str in .schema()")
```

Comment:
this is the failure from within pydantic that I would LOVE to get fixed/addressed at some point. Could either make it `required=False` or do a fancy json schema thing of `anyOf(string, null)` type on output (might cause poor llm performance though)